R/AlabasterParam.R

```diff
@@ -0,0 +1,298 @@
+#'   using the `saveObject()` method of the respective object. If the value for
+#'   the `@qdata` slot is `NULL` the folder *qdata* is not created. At present,
+#'   export of `QFeatures` objects is not supported!
+#' - `@otherData`: `List` data is saved into a folder named *other_data*.
+#' - `@metadata`: `List` data is saved into a filder named *metadata*.
+#'
+#' Note that the data type of the `assays` of imported (previously stored)
+#' `SummarizedExperiment` objects are of type `ReloadedMatrix`.
+#'
+#'
+#' @section On-disk storage for `XcmsExperiment` objects:
+#'
+#' `XcmsExperiment` objects extend the `MsExperiment` object and contain in
+#' addition the results of a preprocessing of the MS data using the *xcms*
+#' package. These objects can be exported/imported in the formats used for
+#' *alabaster*-based storage using the `saveObject()` and `readObject()`
+#' functions as well as using `saveMsObject()` and `readMsObject()` with
+#' an `AlabasterParam` parameter object. As with all other methods, additional
+#' parameters can be passed with the `...` parameter (such as the
+#' `spectraData` parameter for import of a `MsBackendMzR` discussed above).
+#' The storage directory contains all files and folders created by the export
+#' of the `MsExperiment` (see above) and in addition the specific results
+#' of *xcms* from the respective slots of the object:
+#'
+#' - `@chromPeaks`: this numeric matrix is stored in a folder names
+#'   *chrom_peaks*.
+#' - `@chromPeakData`: this `data.frame` is first converted to a `DataFrame`
+#'   and then stored to a folder *chrom_peak_data* (in the *alabaster* format
+#'   for `DataFrame`).
+#' - `@featureDefinitions`: this `data.frame` is first converted to a
+#'   `DataFrame` and then stored to a folder *feature_definitions* (also in
+#'   *alabaster* format for `DataFrame`).
+#' - `@processHistory`: the list of `ProcessHistory` objects is stored in JSON
+#'   format to a file *xcms_experiment_process_history.json*.
+#'
+#'
+#' @author Johannes Rainer, Philippine Louail
+#'
+#' @examples
+#'
+#' ## Export and import a `MsBackendMzR` object:
+#'
+#' library(Spectra)
+#' library(msdata)
+#' fl <- system.file("TripleTOF-SWATH", "PestMix1_DDA.mzML", package = "msdata")
+#' be <- backendInitialize(MsBackendMzR(), fl)
+#' be
+#'
+#' ## Export the object to a temporary directory using the alabaster framework;
+#' ## the equivalent command using the parameter object would be
+#' ## `saveMsObject(be, AlabasterParam(d))`.
+#' d <- file.path(tempdir(), "ms_backend_mzr_example")
+#' saveObject(be, d)
+#'
+#' ## List the content of the folder
+#' dir(d, recursive = TRUE)
+#'
+#' ## The data can be imported again using alabaster's readObject() function
+#' be_in <- readObject(d)
+#' be_in
+#'
+#' ## Alternatively, the data could be restored also using
+#' be_in <- readMsObject(MsBackendMzR(), AlabasterParam(d))
+#'
+#' all.equal(mz(be), mz(be_in))
+#'
+#'
```

Comment:
extra empty line

Reply:
actually there's a few of them throughout the examples, idk if it is on purpose or not :)

Reply:
good point. I wanted to structure the examples a bit and separate the examples for different objects. But maybe better if I separate them with some additional e.g. `####` . I'll check and update.